libs/cart/src/drivers/magento/transforms/outputs/cart-payment.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+
+import { MagentoCartPaymentMethod } from '../../models/outputs/cart-payment-method';
+
+/**
+ * Transforms magento carts into an object usable by daffodil.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffMagentoCartPaymentTransformer {
+
+  /**
+   * Transforms the magento CartPayment from the magento cart query into a DaffCartPaymentMethod.
+   * @param response the response from a magento cart query.
+   */
+  transform(responsePayment: MagentoCartPaymentMethod) {
+    return {
+      method: responsePayment.code
```

Comment:
We should be spreading over the responsePayment like:

```ts
{
 ... responsePayment,
     method: responsePayment.code
}
```
That way we get back a covariant type.

Reply:
Changed in [`5ae0633`](https://github.com/graycoreio/daffodil/pull/654/commits/5ae063386679478ddd92404833faee11dcd8d8dc).